### PR TITLE
Fired events in the theme scanner.

### PR DESCRIPTION
**Pull request by bereznev**
Fired events in ThemeScanner.php class for ability to  extend scan method.

### CODE REVIEW

**Comment by LukeTowers** [comment]
@bereznev please provide a real world use case for both proposed events. Additionally, please add inline event documentation (example here: https://github.com/octobercms/october/blob/develop/modules/backend/classes/Controller.php#L245-L263)

**Comment by bereznev** [comment]
I added an extra scan for the php classes. Using it allows you to add words list using constants (properties) to prevent duplication of these lists. And I need to automatically generate translation files after scanning.

**Comment by LukeTowers** [comment]
Please provide the code you use to listen to the event so I can better understand the purpose.

**Comment by bereznev** [comment]
`        Event::listen('translate.themeScanner.afterScan', function (ThemeScanner $scanner) {
            $dir = themes_path('my-theme/assets/lang');
            if (!is_dir($dir)) {
                File::makeDirectory($dir);
            }
            $exportMode = new MessageExport();
            $result = $exportMode->exportData(['code', 'sk', 'en', 'x',]);

            foreach (['sk', 'en'] as $lang) {
                $content = collect($result)->keyBy('x')->map(function (array $item) use ($lang) {
                    return $item[$lang];
                })->toArray();
                File::put("{$dir}/{$lang}.json", json_encode($content));
            }
        });`

File: classes/ThemeScanner.php [hunk 1]
```diff
@@ -24,7 +25,20 @@ public static function scan()
...
+         * Fires after theme scanning.
+         *
+         * Example usage:
+         *
+         *     Event::listen('translate.themeScanner.afterScan', function (ThemeScanner $scanner) {
+         *         // added an extra scan. Add generation files...
+         *     });
+         *
+         */
+        Event::fire('translate.themeScanner.afterScan', [$obj]);
```
Comment (LukeTowers) [inline classes/ThemeScanner.php:41]: Event should be `rainlab.translate.themeScanner.afterScan`. Also you need to continue to return the result of the scan method

Comment (bennothommo) [reply classes/ThemeScanner.php:41]: @LukeTowers I was going to mention the return thing too, but the method returns `void`, so there was no need for a `return` call in the first place.

**Comment by LukeTowers** [comment]
Change the event names in all the places it's referenced please.

File: README.md [hunk 2]
```diff
@@ -274,6 +274,11 @@ It's possible to translate query string parameters by listening to the `translat
 
 For a possible implementation of the `YourModel::translateParams` method look at the example under `URL parameter translation` from above.
 
+## Extend theme scan
+      Event::listen('rainlab.translate.afterScan', function (ThemeScanner $scanner) {
```
Comment (LukeTowers) [inline README.md:278]: Event name still needs to be changed here. Please also add a newline between the header and the code example.

**Comment by LukeTowers** [comment]
LGTM

**Pull request merged**